### DISCUSSION TITLE
Custom User-Agent header and/or other headers for back-channel request

### DISCUSSION THREAD

**Discussion by studersi** [OpenIDC/mod_auth_openidc#1151] [Ideas]
When debugging issues with mod_auth_openidc and mod_oauth2, we sometimes struggle with request correlation.

It would be helpful to set custom user-agent string for back-channel request to associate them with a particular Apache instance.

Further, it could be helpful to add custom tracing headers like the traceparent header ([Trace Context - Traceparent Header](https://www.w3.org/TR/trace-context/#traceparent-header) to back-channel request for tracing the access across all services that are involved in processing a request.

**Comment by zandbelt** [comment]
yes, see: https://github.com/OpenIDC/mod_auth_openidc/commit/e8d5adcf82f449f1af0fd234b770d32cec61a5e1

**Reply by studersi** [reply]
Thanks for the link. From what I see, this adds additional information about the versions of the various involved technologies. However, what we would be interested in is extending this information with tracing IDs. Would this be possible somehow?

Consider the following case:
* Client sends a request to Apache
* Apache adds a traceparent header
* mod_auth_openidc/mod_oauth2 sends a request to the IDP with the newly created traceparent header
* On the IDP we see the traceparent header and can easily correlate it with the original request from the client to Apache

From what I can tell, this is not possible with the changes in https://github.com/OpenIDC/mod_auth_openidc/commit/e8d5adcf82f449f1af0fd234b770d32cec61a5e1.

Consider also the following case:
We have a lot of request from mod_auth_openidc/mod_oauth2 to the IDP. However, there are multiple, separate Apache instances running on the same host, initiating requests to the IDP. If we could extend the User-Agent (or add another header) with information about which Apache instance originally received the request, we could determine much more easily which Apache instance(s) a set of problematic requests originates from.

**Reply by zandbelt** [reply]
ah, yes, I was too quick; we'll take a look and let you know

**Reply by zandbelt** [reply]
see https://github.com/OpenIDC/mod_auth_openidc/commit/0d1e17c504f10a813c5a9fd134b8717b0f336ba4

**Comment by zandbelt** [comment]
this is now in 2.4.15 https://github.com/OpenIDC/mod_auth_openidc/releases/tag/v2.4.15
